hackathon_site/event/test_api.py

```diff
@@ -52,42 +52,77 @@ def test_user_has_profile(self):
         self.assertEqual(response.json(), serializer.data)
 
 
-class CurrentTeamTestCase(SetupUserMixin, APITestCase):
+class JoinTeamTestCase(SetupUserMixin, APITestCase):
     def setUp(self):
         super().setUp()
         self.group = Group.objects.create(name="Test Users")
         self.user.groups.add(self.group)
         self.team = Team.objects.create()
 
         self.profile = Profile.objects.create(user=self.user, team=self.team)
+        self.team_code = self.team.team_code
+        self.view_name = "api:event:join-team"
 
-        self.view = reverse("api:event:current-team")
+    def _build_view(self, team_code):
+        return reverse(self.view_name, kwargs={"team_code": team_code})
+
+    def test_invalid_key(self):
+        self._login()
+        response = self.client.post(self._build_view("56ABD"))
+        self.assertEqual(response.status_code, status.HTTP_400_BAD_REQUEST)
+
+    def test_join_full_team(self):
+        self._login()
+        team = self._make_full_event_team(self_users=False)
+        print(team.team_code)
+        response = self.client.post(self._build_view(team.team_code))
+        self.assertEqual(str(response.content), 'b\'{"detail":"Team is full"}\'')
```

Comment:
Look at some of our other API tests. You can do `data = response.json()` to get a dict, then compare that dict rather than an ugly string like this

Reply:
done